Kernel/LPO.cpp

```diff
@@ -196,30 +196,26 @@ Ordering::Result LPO::lpo(TermList tl1, TermList tl2) const
 {
   CALL("LPO::lpo(TermList, TermList)");
 
+  if(tl1==tl2) {
```

Comment:
This check is not in the original specification but the original did not use term sharing either... Does this make sense here? Otherwise we have to return `EQUAL` in `lexMAE` as I wrote in the other thread.

Reply:
I think so! It should be a fairly fast check and it saves us recursing through terms uselessly. `return GREATER` at the end of `lexMAE` is confusing and looks like a bug though - could we safely make it `ASSERTION_VIOLATION` instead, with a comment explaining things? We know statically that `lexMAE` is only called from `lpo` after the equality check. Should also be a slight performance win as the compiler will know that it doesn't have to generate code for the loop exit (or loop condition, come to think of it).